File: TiendaVirtual/TiendaVirtualBackEnd/Model/CategoriaRequest.cs
using System.ComponentModel.DataAnnotations;

namespace Model
{
    public class CategoriaRequest
    {
        public int Id { get; set; }

        [StringLength(40)] // ==> en base  de datos VARCHAR(40)
        public string? Nombre { get; set; }

        [StringLength(150)]
        public string? Descripcion { get; set; }
        public bool? Estado { get; set; }

        [StringLength(5)]
        public string? Codigo { get; set; }

        //[InverseProperty("IdCategoriaNavigation")]
        //public virtual ICollection<Producto> Productos { get; } = new List<Producto>();

        //[InverseProperty("IdCategoriaNavigation")]
        //public virtual ICollection<VentaDetalle> VentaDetalles { get; } = new List<VentaDetalle>();
    }
}

File: TiendaVirtual/TiendaVirtualFrontEnd/src/app/models/categoria-request.ts
export class CategoriaRequest {
    id: number;
    nombre: string | null;
    descripcion: string | null;
    estado: boolean | null;
    codigo: string | null;
    constructor() {
        this.id = 0;
        this.nombre = "";
        this.descripcion = "";
        this.estado = true;
        this.codigo = "";
    }
}